src/svc/crd/metabase.rs

```diff
@@ -0,0 +1,610 @@
+                    "Override plan for custom resource",
+                );
+
+                let oplan = modified.spec.instance.plan.to_owned();
+                plan.id.clone_into(&mut modified.spec.instance.plan);
+
+                debug!(
+                    kind = &kind,
+                    namespace = &namespace,
+                    name = &name,
+                    "Update information of custom resource",
+                );
+
+                let patch = resource::diff(&*origin, &modified).map_err(ReconcilerError::Diff)?;
+                let modified =
+                    resource::patch(kube.to_owned(), &modified, patch.to_owned()).await?;
+
+                let action = &Action::OverridesInstancePlan;
+                let message = &format!("Overrides instance plan from '{}' to '{}'", oplan, plan.id);
+
+                info!(
+                    action = action.to_string(),
+                    kind = &kind,
+                    namespace = &namespace,
+                    name = &name,
+                    message = message,
+                    "Create event for custom resource",
+                );
+
+                recorder::normal(kube.to_owned(), &modified, action, message).await?;
+            }
+
+            // Stop reconciliation here and wait for next iteration, already
+            // triggered by the above patch request
+            return Ok(());
+        }
+
+        // ---------------------------------------------------------------------
+        // Step 3: upsert addon
+
+        info!(
+            kind = &kind,
+            namespace = &namespace,
+            name = &name,
+            "Upsert addon for custom resource",
+        );
+
+        let addon = modified.upsert(&apis).await?;
+
+        modified.set_addon_id(Some(addon.id.to_owned()));
+
+        debug!(
+            kind = &kind,
+            namespace = &namespace,
+            name = &name,
+            "Update information and status of custom resource",
+        );
+
+        let patch = resource::diff(&*origin, &modified).map_err(ReconcilerError::Diff)?;
+        let mut modified = resource::patch(kube.to_owned(), &modified, patch.to_owned())
+            .and_then(|modified| resource::patch_status(kube.to_owned(), modified, patch))
+            .await?;
+
+        let action = &Action::UpsertAddon;
+        let message = &format!(
+            "Create managed postgresql instance on clever-cloud '{}'",
+            addon.id
+        );
+        recorder::normal(kube.to_owned(), &modified, action, message).await?;
+
+        // ---------------------------------------------------------------------
+        // Step 4: upsert secret
+
+        let secrets = modified.secrets(&apis).await?;
+
+        // capture the url to update the status un Step 5
+        // this is needed because `Secret` takes ownership of the secrets
+        // alternatively we can just clone the whole `BTreeMap`
+        let url = secrets
```

Comment:
Use match syntax here, you need to clone only once the value. Besides the value, should be `METABASE_URL`